crates/language/src/language.rs

```diff
@@ -170,156 +136,149 @@ pub trait NodeTypes {
     fn node_types(&self) -> &[Vec<Field>];
 }
 
-#[enum_dispatch(TargetLanguage)]
-pub trait Language: NodeTypes {
-    /// tree sitter language to parse the source
-    fn get_ts_language(&self) -> &TSLanguage;
-
-    fn language_name(&self) -> &'static str;
-
-    fn skip_snippet_compilation_of_field(&self, _sort_id: SortId, _field_id: FieldId) -> bool {
-        false
-    }
+#[derive(Clone, Debug)]
+pub struct Tree {
```

Comment:
The `Tree` data structure encapsulates a TreeSitter tree, while implementing the `Ast` trait. That one just has a single method to get the root node as an `AstNode`.